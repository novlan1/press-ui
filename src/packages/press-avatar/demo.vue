<template>
  <div class="demo-wrap">
    <demo-block :title="t('basicUsage')">
      <PressAvatar :src="src" />
    </demo-block>

    <demo-block :title="t('shape')">
      <PressAvatar
        :src="src"
        round
        :custom-style="customStyle"
      />
      <PressAvatar
        :src="src"
        :round="false"
      />
    </demo-block>

    <demo-block :title="t('size')">
      <PressAvatar
        :src="src"
        size="30"
        :custom-style="customStyle"
      />
      <PressAvatar
        :src="src"
        size="40"
        :custom-style="customStyle"
      />
      <PressAvatar
        :src="src"
        size="50"
      />
    </demo-block>

    <demo-block :title="t('icon')">
      <PressAvatar
        :src="src"
        icon="star-o"
        :custom-style="customStyle"
      />
      <PressAvatar
        :src="src"
        icon="like-o"
      />
    </demo-block>

    <demo-block :title="t('text')">
      <PressAvatar
        text="Y"
        :custom-style="customStyle"
      />
      <PressAvatar
        text="G"
        random-background-color
        :custom-style="customStyle"
      />
      <PressAvatar
        text="杨"
        random-background-color
      />
    </demo-block>

    <demo-block :title="t('error')">
      <PressAvatar />
    </demo-block>

    <demo-block :title="t('avatarGroup')">
      <PressAvatarGroup
        :urls="urls"
        custom-style="margin-bottom: 16px"
        @showMore="showMore"
      />

      <PressAvatarGroup
        :urls="urls"
        gap="0.6"
        custom-style="margin-bottom: 16px"
        @showMore="showMore"
      />

      <PressAvatarGroup
        :urls="urls"
        size="50"
        custom-style="margin-bottom: 16px"
        @showMore="showMore"
      />
    </demo-block>
  </div>
</template>
<script>
import PressAvatarGroup from 'press-ui/press-avatar-group/press-avatar-group.vue';
import PressAvatar from 'press-ui/press-avatar/press-avatar.vue';


const src = 'https://mike-1255355338.cos.ap-guangzhou.myqcloud.com/article/2023/10/own_mike_0ab8e66284a3617ed9.jpeg';


export default {
  i18n: {
    'zh-CN': {
      shape: '头像形状',
      size: '头像尺寸',
      icon: '图标头像',
      text: '文字头像',
      error: '加载失败',
      avatarGroup: '头像组',
    },
    'en-US': {
      shape: 'Shape',
      size: 'Size',
      icon: 'Icon',
      text: 'Text',
      error: 'Error',
      avatarGroup: 'Avatar Group',
    },
  },
  components: {
    PressAvatar,
    PressAvatarGroup,
  },
  data() {
    return {
      src,
      customStyle: 'margin-right: 20px',
      urls: [
        src,
        'https://mike-1255355338.cos.ap-guangzhou.myqcloud.com/article/2023/10/own_mike_ef8e6956717db34745.jpg',
        src,
        'https://mike-1255355338.cos.ap-guangzhou.myqcloud.com/article/2023/10/own_mike_7157200e469d0468b3.jpg',
        src,
        'https://mike-1255355338.cos.ap-guangzhou.myqcloud.com/article/2023/10/own_mike_a17f94292a34e9f644.jpeg',
        src,
      ],
    };
  },
  methods: {
    showMore() {
      this.onGTip('[showMore]');
    },
  },
};
</script>
<style scoped lang="scss">
</style>
