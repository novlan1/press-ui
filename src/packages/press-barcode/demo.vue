<template>
  <div class="demo-wrap demo-wrap--gray">
    <div class="barcode-wrap">
      <PressBarcode
        :value="value"
        :width="options.width"
        :height="options.height"
        :format="options.format"
        :font-size="options.fontSize"
      />
    </div>

    <DemoInput
      :autosize="autosize"
      :value="value"
      @confirm="onConfirmInput"
    />
  </div>
</template>
<script>
import PressBarcode from 'press-ui/press-barcode/press-barcode.vue';
import DemoInput from 'press-ui/press-demo-input/press-demo-input.vue';


export default {
  components: {
    PressBarcode,
    DemoInput,
  },
  data() {
    return {
      autosize: { maxHeight: 600, minHeight: 50 },
      value: '123123123123',
      options: {
        width: 1,
        height: 50,
        fontSize: 12,
        format: 'CODE128',
      },
    };
  },
  methods: {
    onConfirmInput(value) {
      this.value = value;
    },
  },
};
</script>
<style scoped lang="scss">
.barcode-wrap {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
}
</style>
