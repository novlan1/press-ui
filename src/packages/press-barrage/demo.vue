<template>
  <div class="demo-wrap">
    <demo-block
      :title="t('basicUsage')"
    >
      <PressBarrage v-model="list">
        <div
          class="video"
          style="width: 100%; height: 150px; background: #f2f3f5;"
        />
      </PressBarrage>

      <PressButton
        type="primary"
        size="small"
        custom-style="margin-top: 10px;"
        @click="add"
      >
        弹幕
      </PressButton>
    </demo-block>

    <demo-block
      :title="t('videoBarrage')"
    >
      <PressBarrage
        ref="pressBarrage"
        v-model="videoList"
        :auto-play="false"
      >
        <div
          class="video"
          style="width: 100%; height: 150px; background: #f2f3f5;"
        />
      </PressBarrage>

      <div>
        <PressButton
          type="primary"
          size="small"
          :disabled="!isPlay"
          custom-style="margin-top: 10px;margin-right: 10px;"
          @click="videoAdd"
        >
          弹幕
        </PressButton>

        <PressButton
          type="default"
          size="small"
          custom-style="margin-top: 10px;"
          @click="toggle"
        >
          {{ isPlay ? t('pause') : t('play') }}
        </PressButton>
      </div>
    </demo-block>
  </div>
</template>
<script>
import { setAdapter } from 'press-ui/common/vue3/set';
import PressBarrage from 'press-ui/press-barrage/press-barrage.vue';
import PressButton from 'press-ui/press-button/press-button.vue';


export default {
  i18n: {
    'zh-CN': {
      barrage: '弹幕',
      play: '开始',
      pause: '暂停',
      videoBarrage: '模仿视频弹幕',
      lightweight: '轻量',
      customizable: '可定制的',
      mobile: '移动端',
      library: '组件库',
    },
    'en-US': {
      barrage: 'barrage',
      play: 'play',
      pause: 'pause',
      videoBarrage: 'Imitate video barrage',
      lightweight: 'Lightweight',
      customizable: 'Customizable',
      mobile: 'Mobile',
      library: 'Library',
    },
  },
  components: {
    PressBarrage,
    PressButton,
  },
  data() {
    const defaultList = [
      { id: 100, text: this.t('lightweight') },
      { id: 101, text: this.t('customizable') },
      { id: 102, text: this.t('mobile') },
      { id: 103, text: 'Vue' },
      { id: 104, text: this.t('library') },
      { id: 105, text: 'PressUI' },
      { id: 106, text: '666' },
    ];

    return {
      list: defaultList,
      videoList: [
        ...defaultList,
      ],
      isPlay: false,

    };
  },
  watch: {
    isPlay: {
      handler(val) {
        if (val) {
          this.$refs.pressBarrage?.play();
        } else {
          this.$refs.pressBarrage?.pause();
        }
      },
    },
  },
  methods: {
    add() {
      setAdapter(this, 'list', [
        ...this.list,
        {
          id: Math.random(),
          text: 'Barrage',
        },
      ]);
    },
    videoAdd() {
      setAdapter(this, 'videoList', [
        ...this.videoList,
        {
          id: Math.random(),
          text: 'Barrage',
        },
      ]);
    },
    toggle() {
      this.isPlay = !this.isPlay;
    },
  },
};
</script>
<style scoped lang="scss">
</style>
