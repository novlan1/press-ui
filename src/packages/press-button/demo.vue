<template>
  <div class="demo-wrap">
    <demo-block :title="t('type')">
      <div class="demo-button-row">
        <PressButton
          type="default"
          :custom-style="customStyle"
          @click="onClick"
        >
          {{ t('default') }}
        </PressButton>
        <PressButton
          type="primary"
          :custom-style="`${customStyle}`"
          @click="onVibrateShort"
        >
          {{ t('primary') }}
        </PressButton>
        <PressButton
          type="info"
          @click="onVibrateLong"
        >
          {{ t('info') }}
        </PressButton>
      </div>

      <div class="demo-button-row">
        <PressButton
          type="warning"
          :custom-style="customStyle"
          @click="onLaunchMp"
        >
          {{ t('warning') }}
        </PressButton>
        <PressButton
          type="danger"
          @click="onLaunchMp(1)"
        >
          {{ t('danger') }}
        </PressButton>
      </div>
    </demo-block>

    <demo-block :title="t('plain')">
      <div class="demo-button-row">
        <PressButton
          plain
          type="default"
          :custom-style="customStyle"
        >
          {{ t('plain') }}
        </PressButton>
        <PressButton
          plain
          type="primary"
          :custom-style="customStyle"
        >
          {{ t('plain') }}
        </PressButton>
      </div>

      <PressButton
        plain
        type="info"
      >
        {{ t('plain') }}
      </PressButton>
    </demo-block>


    <demo-block :title="t('hairline')">
      <PressButton
        plain
        hairline
        type="primary"
        :custom-style="customStyle"
      >
        {{ t('hairlineButton') }}
      </PressButton>
      <PressButton
        plain
        hairline
        type="info"
      >
        {{ t('hairlineButton') }}
      </PressButton>
    </demo-block>

    <demo-block :title="t('disabled')">
      <div class="demo-button-row">
        <PressButton
          disabled
          type="primary"
          :custom-style="customStyle"
        >
          {{ t('disabled') }}
        </PressButton>
        <PressButton
          disabled
          type="info"
          :custom-style="customStyle"
        >
          {{ t('disabled') }}
        </PressButton>
      </div>

      <PressButton
        disabled
        type="primary"
        :custom-style="customStyle"
      >
        {{ t('disabled') }}
      </PressButton>
    </demo-block>

    <demo-block :title="t('loading')">
      <PressButton
        loading
        type="primary"
        :custom-style="customStyle"
      />
      <PressButton
        loading
        type="primary"
        loading-type="spinner"
        :custom-style="customStyle"
      />
      <PressButton
        loading
        loading-type="circular-tdesign"
        type="info"
        :loading-text="t('loadingText')"
      />
    </demo-block>

    <demo-block :title="t('shape')">
      <PressButton
        square
        type="primary"
        :custom-style="customStyle"
      >
        {{ t('square') }}
      </PressButton>
      <PressButton
        round
        type="info"
      >
        {{ t('round') }}
      </PressButton>
    </demo-block>


    <demo-block :title="t('icon')">
      <PressButton
        icon="star-o"
        type="primary"
        :custom-style="customStyle"
      />
      <PressButton
        icon="star-o"
        type="primary"
        :custom-style="customStyle"
      >
        {{ t('button') }}
      </PressButton>
      <PressButton
        icon="https://mike-1255355338.cos.ap-guangzhou.myqcloud.com/press%2Fimg%2Fpress-ui-avatar-transparent-2.png"
      >
        {{ t('button') }}
      </PressButton>
    </demo-block>

    <demo-block :title="t('size')">
      <div class="demo-button-row">
        <PressButton
          type="primary"
          size="large"
        >
          {{ t('large') }}
        </PressButton>
      </div>

      <PressButton
        type="primary"
        size="normal"
        :custom-style="customStyle"
      >
        {{ t('normal') }}
      </PressButton>
      <PressButton
        type="primary"
        size="small"
        :custom-style="customStyle"
      >
        {{ t('small') }}
      </PressButton>
      <PressButton
        type="primary"
        size="mini"
      >
        {{ t('mini') }}
      </PressButton>
    </demo-block>

    <demo-block :title="t('blockElement')">
      <PressButton
        type="primary"
        block
      >
        {{ t('blockElement') }}
      </PressButton>
    </demo-block>

    <demo-block :title="t('customColor')">
      <PressButton
        color="#7232dd"
        :custom-style="customStyle"
        open-type="openSetting"
      >
        {{ t('pure') }}
      </PressButton>
      <PressButton
        color="#7232dd"
        plain
        :custom-style="customStyle"
        open-type="feedback"
      >
        {{ t('pure') }}
      </PressButton>
      <PressButton
        color="linear-gradient(to right, #4bb0ff, #6149f6)"
        open-type="share"
      >
        {{ t('gradient') }}
      </PressButton>
    </demo-block>

    <demo-block
      v-for="(item,index) of [0,1,2]"
      :key="index"
      :title="index===1 ?'E-SPORT-DISABLE' : index===2 ? 'E-SPORT-LOADING' : 'E-SPORT'"
    >
      <div class="demo-button-row">
        <!-- :size="{width: '136px', height: '36px'}" -->
        <PressButton
          type="e-sport-primary"
          :disabled="index===1"
          :loading="index===2"
          :custom-style="customStyle"
        >
          primary
        </PressButton>
        <PressButton
          type="e-sport-primary-bg"
          :disabled="index===1"
          :loading="index===2"
          :custom-style="customStyle"
        >
          primary-bg
        </PressButton>
      </div>

      <div class="demo-button-row">
        <PressButton
          type="e-sport-primary-bg-lg"
          :disabled="index===1"
          :loading="index===2"
          :custom-style="customStyle"
        >
          primary-bg-lg
        </PressButton>
      </div>

      <div class="demo-button-row">
        <PressButton
          type="e-sport-primary-bg-xl"
          :disabled="index===1"
          :loading="index===2"
          :custom-style="customStyle"
        >
          primary-bg-xl
        </PressButton>
      </div>

      <div class="demo-button-row">
        <PressButton
          type="e-sport-secondary"
          :disabled="index===1"
          :loading="index===2"
          :custom-style="customStyle"
          :size="{width: '136px', height: '36px'}"
        >
          secondary
        </PressButton>

        <!-- 线框的loader图片是透明的，没效果，直接隐藏 -->
        <PressButton
          v-if="index!==2"
          type="e-sport-border"
          :disabled="index===1"
          :loading="index===2"
          :custom-style="customStyle"
          :size="{width: '136px', height: '36px'}"
        >
          border
        </PressButton>
      </div>
    </demo-block>
  </div>
</template>
<script>
import PressButton from 'press-ui/press-button/press-button.vue';
import { isNotInUni } from 'press-ui/common/utils/utils';
import { launchMiniProgramInGame } from 't-comm/lib/msdk/mini-program';


export default {
  i18n: {
    'zh-CN': {
      type: '按钮类型',
      size: '按钮尺寸',
      icon: '图标按钮',
      loading: '加载状态',
      disabled: '禁用状态',
      shape: '按钮形状',
      default: '默认按钮',
      primary: '主要按钮',
      info: '信息按钮',
      danger: '危险按钮',
      warning: '警告按钮',
      large: '大号按钮',
      normal: '普通按钮',
      small: '小型按钮',
      mini: '迷你按钮',
      plain: '朴素按钮',
      square: '方形按钮',
      round: '圆形按钮',
      hairline: '细边框',
      hairlineButton: '细边框按钮',
      loadingText: '加载中...',
      router: '页面导航',
      urlRoute: 'URL 跳转',
      vueRoute: '路由跳转',
      customColor: '自定义颜色',
      pure: '单色按钮',
      gradient: '渐变色按钮',
      blockElement: '块级元素',
    },
    'en-US': {
      type: 'Type',
      size: 'Size',
      icon: 'Icon',
      loading: 'Loading',
      disabled: 'Disabled',
      shape: 'Shape',
      default: 'Default',
      primary: 'Primary',
      info: 'Info',
      danger: 'Danger',
      warning: 'Warning',
      large: 'Large',
      normal: 'Normal',
      small: 'Small',
      mini: 'Mini',
      plain: 'Plain',
      square: 'Square',
      round: 'Round',
      hairline: 'Hairline',
      hairlineButton: 'Hairline',
      loadingText: 'Loading...',
      router: 'Router',
      urlRoute: 'URL',
      vueRoute: 'Vue Router',
      customColor: 'Custom Color',
      pure: 'Pure',
      gradient: 'Gradient',
      blockElement: 'Block Element',
    },
  },
  components: {
    PressButton,
  },
  data() {
    return {
      customStyle: 'margin-right: 16px;',
    };
  },
  onReady() {

  },
  methods: {
    onLaunchMp(isQQ) {
      console.log('[launchMiniProgramInGame]');
      // #ifdef H5
      launchMiniProgramInGame({
        appId: 'wx99c1c10a389e7433',
        isWxMp: !isQQ,
      });
      // #endif
    },
    onClick(event) {
      console.log('[CLICK]', event);
    },
    onVibrateLong() {
      if (isNotInUni()) return;

      uni.vibrateLong({
        success() {
          console.log('success');
        },
        fail() {
          console.log('fail');
        },
      });
    },
    onVibrateShort() {
      // #ifdef H5
      navigator.vibrate(1000);
      // #endif

      if (isNotInUni()) return;

      uni.vibrateLong({
        success() {
          console.log('success');
        },
        fail() {
          console.log('fail');
        },
      });
    },
  },
};
</script>

<style scoped lang="scss">
.demo-button-row {
  margin-bottom: 12px;
}
</style>

