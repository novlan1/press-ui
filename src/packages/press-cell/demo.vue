<template>
  <div class="demo-wrap demo-wrap--gray">
    <demo-block
      :title="t('basicUsage')"
      :header-style="headerStyle"
      :section-style="sectionStyle"
    >
      <PressCellGroup>
        <PressCell
          :title="t('cell')"
          :value="t('content')"
        />
        <PressCell
          :title="t('cell')"
          :value="t('content')"
          :label="t('desc')"
          :border="false"
        />
      </PressCellGroup>
    </demo-block>

    <demo-block
      :title="t('insetGrouped')"
      :section-style="sectionStyle"
      :header-style="headerStyle"
    >
      <PressCellGroup inset>
        <PressCell
          :title="t('cell')"
          :value="t('content')"
        />
        <PressCell
          :title="t('cell')"
          :value="t('content')"
          :label="t('desc')"
        />
      </PressCellGroup>
    </demo-block>

    <demo-block
      :title="t('largeSize')"
      :section-style="sectionStyle"
      :header-style="headerStyle"
    >
      <PressCell
        :title="t('cell')"
        :value="t('content')"
        size="large"
      />
      <PressCell
        :title="t('cell')"
        :value="t('content')"
        size="large"
        :label="t('desc')"
      />
    </demo-block>

    <demo-block
      :title="t('showIcon')"
      :section-style="sectionStyle"
      :header-style="headerStyle"
    >
      <PressCell
        :title="t('cell')"
        icon="location-o"
      />
    </demo-block>

    <demo-block
      :title="t('showArrow')"
      :section-style="sectionStyle"
      :header-style="headerStyle"
    >
      <PressCell
        :title="t('cell')"
        is-link
      />
      <PressCell
        :title="t('cell')"
        :value="t('content')"
        is-link
      />
      <PressCell
        :title="t('cell')"
        :value="t('content')"
        is-link
        arrow-direction="down"
      />
    </demo-block>

    <demo-block
      :title="t('router')"
      :section-style="sectionStyle"
      :header-style="headerStyle"
    >
      <PressCell
        is-link
        :title="t('cell')"
        link-type="navigateTo"
        url="/pages/press/col/col"
      />
    </demo-block>

    <demo-block
      :title="t('groupTitle')"
      :section-style="sectionStyle"
      :header-style="headerStyle"
    >
      <PressCellGroup :title="`${t('group')}1`">
        <PressCell
          :title="t('cell')"
          :value="t('content')"
        />
      </PressCellGroup>
      <PressCellGroup :title="`${t('group')}2`">
        <PressCell
          :title="t('cell')"
          :value="t('content')"
        />
      </PressCellGroup>
    </demo-block>

    <demo-block
      :title="t('useSlots')"
      :section-style="sectionStyle"
      :header-style="headerStyle"
    >
      <PressCell
        :value="t('content')"
        is-link
        :use-title-slot="true"
      >
        <template #title>
          <span class="press-cell-text">
            {{ t('cell') }}
          </span>
          <PressTag type="danger">
            {{ t('tag') }}
          </PressTag>
        </template>
      </PressCell>
      <PressCell
        :title="t('cell')"
        icon="shop-o"
      >
        <template #right-icon>
          <PressIconPlus
            name="search"
            class="custom-icon"
          />
        </template>
      </PressCell>
    </demo-block>

    <demo-block
      :title="t('verticalCenter')"
      :section-style="sectionStyle"
      :header-style="headerStyle"
    >
      <PressCell
        center
        :title="t('cell')"
        :value="t('content')"
        :label="t('desc')"
      />
    </demo-block>

    <demo-block
      title="E-SPORT"
      :header-style="headerStyle"
      :section-style="sectionStyle"
    >
      <div class="demo--e-sport-wrap">
        <PressCell
          :title="t('matchMode')"
          :value="t('knockout')"
          type="e-sport"
          is-link
          :border="false"
        />
        <PressCell
          :title="t('wayToWin')"
          :value="t('defaultBo1')"
          type="e-sport"
          is-link
          :border="false"
        />
        <PressCell
          :title="t('allowModifyAvatar')"
          :border="false"
          type="e-sport"
        >
          <template #right-icon>
            <PressSwitch
              custom-class="press-switch--e-sport"
              :checked="switchChecked[1]"
              @change="val=>onChangeSwitch(val, 1)"
            />
          </template>
        </PressCell>

        <PressCell
          :title="t('getPhoneNumber')"
          :label="t('getPhoneNumberDesc')"
          type="e-sport"
          center
          :border="false"
        >
          <template #right-icon>
            <PressSwitch
              custom-class="press-switch--e-sport"
              :checked="switchChecked[2]"
              @change="val=>onChangeSwitch(val, 2)"
            />
          </template>
        </PressCell>
      </div>
    </demo-block>
  </div>
</template>
<script>
import PressCellGroup from 'press-ui/press-cell-group/press-cell-group.vue';
import PressCell from 'press-ui/press-cell/press-cell.vue';
import PressIconPlus from 'press-ui/press-icon-plus/press-icon-plus.vue';

import PressSwitch from 'press-ui/press-switch/press-switch.vue';
import PressTag from 'press-ui/press-tag/press-tag.vue';


export default {
  i18n: {
    'zh-CN': {
      cell: '单元格',
      valueOnly: '只设置 value',
      showIcon: '展示图标',
      showArrow: '展示箭头',
      largeSize: '单元格大小',
      group: '分组',
      groupTitle: '分组标题',
      router: '页面跳转',
      urlRoute: 'URL 跳转',
      vueRoute: '路由跳转',
      useSlots: '使用插槽',
      insetGrouped: '卡片风格',
      verticalCenter: '垂直居中',

      matchMode: '比赛模式',
      knockout: '淘汰赛',
      defaultBo1: '默认一局胜负',
      allowModifyAvatar: '允许比赛中修改队名队徽',
      getPhoneNumber: '获取参赛者手机号码',
      getPhoneNumberDesc: '便于发放奖励和联系参赛者，可在队员信息中查看',
    },
    'en-US': {
      cell: 'Cell title',
      valueOnly: 'Value only',
      showIcon: 'Left Icon',
      showArrow: 'Link',
      largeSize: 'Size',
      group: 'Group',
      groupTitle: 'Group Title',
      router: 'Router',
      urlRoute: 'URL',
      vueRoute: 'Vue Router',
      useSlots: 'Use Slots',
      insetGrouped: 'Inset Grouped',
      verticalCenter: 'Vertical center',

      matchMode: 'Match Mode',
      knockout: 'Knockout',
      defaultBo1: 'Default Bo1',
      allowModifyAvatar: 'Allow modification of team info',
      getPhoneNumber: 'Obtain player\'s phone number',
      getPhoneNumberDesc: 'Distribute rewards and contact participants',
    },
  },
  components: {
    PressSwitch,
    PressCell,
    PressCellGroup,
    PressTag,
    PressIconPlus,
  },
  data() {
    return {
      sectionStyle: 'margin: 0;background: #f7f8fa;',
      headerStyle: 'background: #f7f8fa;',
      switchChecked: {
        1: false,
        2: false,
      },
    };
  },
  methods: {
    onChangeSwitch(value, number) {
      this.switchChecked[number] = value;
    },
  },
};
</script>
<style scoped lang="scss">
.demo-wrap {
  .press-cell-text {
    margin-right: 4px;
    vertical-align: middle;
  }
}
</style>
