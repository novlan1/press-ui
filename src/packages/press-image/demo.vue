<template>
  <div class="demo-wrap">
    <demo-block :title="t('basicUsage')">
      <PressImage
        :show-loading="true"
        :src="src"
        width="100"
        height="100"
        @click="click"
      />
    </demo-block>

    <demo-block :title="t('mode')">
      <PressRow
        v-for="(row, rowIndex) of MODE_LIST"
        :key="getIndex(rowIndex, 'row')"
      >
        <PressCol
          v-for="(col, colIndex) of row"
          :key="getIndex(colIndex, 'col')"
          :span="col.span"
        >
          <div class="demo-grid">
            <PressImage
              :show-loading="true"
              width="100"
              height="100"
              :src="src"
              :mode="col.mode"
              @click="click"
            />
            <span
              class="image-title"
              :class="col.titleClass"
            >{{ col.mode }}</span>
          </div>
        </PressCol>
      </PressRow>
    </demo-block>

    <demo-block :title="t('round')">
      <PressRow
        v-for="(row, rowIndex) of MODE_LIST"
        :key="getIndex(rowIndex, 'row')"
      >
        <PressCol
          v-for="(col, colIndex) of row"
          :key="getIndex(colIndex, 'col')"
          :span="col.span"
        >
          <div class="demo-grid">
            <PressImage
              :show-loading="true"
              width="100"
              height="100"
              :src="src"
              round
              :mode="col.mode"
              @click="click"
            />
            <span
              class="image-title"
              :class="col.titleClass"
            >{{ col.mode }}</span>
          </div>
        </PressCol>
      </PressRow>
    </demo-block>

    <demo-block :title="t('loading')">
      <PressRow>
        <PressCol
          :span="8"
        >
          <div class="demo-grid">
            <PressImage
              :show-loading="true"
              width="100"
              height="100"
            />

            <span class="image-title">{{ t('defaultTip') }}</span>
          </div>
        </PressCol>

        <PressCol
          :span="8"
        >
          <div class="demo-grid">
            <PressImage
              :show-loading="true"
              width="100"
              height="100"
            >
              <template #loading>
                <PressLoadingPlus
                  type="spinner"
                  size="20"
                />
              </template>
            </PressImage>

            <span class="image-title">{{ t('customTip') }}</span>
          </div>
        </PressCol>
      </PressRow>
    </demo-block>


    <demo-block :title="t('error')">
      <PressRow>
        <PressCol
          :span="8"
        >
          <div class="demo-grid">
            <PressImage
              :show-loading="true"
              src="src"
              width="100px"
              height="100px"
            />
            <span class="image-title">{{ t('defaultTip') }}</span>
          </div>
        </PressCol>

        <PressCol
          :span="8"
        >
          <div class="demo-grid">
            <PressImage
              :show-loading="true"
              src="src"
              width="100px"
              height="100px"
            >
              <template #error>
                <span
                  style="font-size: 14px;"
                >加载失败</span>
              </template>
            </PressImage>
            <span
              class="image-title"
            >{{ t('customTip') }}</span>
          </div>
        </PressCol>
      </PressRow>
    </demo-block>

    <ImagePreview
      ref="imagePreviewRef"
      mode="functional"
    />
  </div>
</template>
<script>
import PressCol from 'press-ui/press-col/press-col.vue';
import { imagePreviewHandler } from 'press-ui/press-image-preview/index';

import ImagePreview from 'press-ui/press-image-preview/press-image-preview.vue';
import PressImage from 'press-ui/press-image/press-image.vue';

import PressLoadingPlus from 'press-ui/press-loading-plus/press-loading-plus.vue';
import PressRow from 'press-ui/press-row/press-row.vue';


const MODE_LIST = [
  [
    {
      mode: 'aspectFit',
      titleClass: 'image-title--gap',
      span: 8,
    },
    {
      mode: 'aspectFill',
      titleClass: 'image-title--gap',
      span: 8,
    },
    {
      mode: 'center',
      titleClass: '',
      span: 8,
    },
  ],
  [
    {
      mode: 'widthFix',
      titleClass: 'image-title--gap',
      span: 8,
    },
    {
      mode: 'heightFix',
      titleClass: 'image-title--long',
      span: 16,
    },
  ],
];


export default {
  i18n: {
    'zh-CN': {
      error: '加载失败提示',
      loading: '加载中提示',
      defaultTip: '默认提示',
      customTip: '自定义提示',
      mode: '模式',
      round: '圆形',
    },
    'en-US': {
      error: 'Error',
      loading: 'Loading',
      defaultTip: 'Default',
      customTip: 'Custom',
      mode: 'Mode',
      round: 'Round',
    },
  },
  components: {
    PressCol,
    PressRow,
    PressImage,
    PressLoadingPlus,
    ImagePreview,
  },
  data() {
    return {
      src: 'https://mike-1255355338.cos.ap-guangzhou.myqcloud.com/article/2023/10/own_mike_a17f94292a34e9f644.jpeg',
      MODE_LIST,
    };
  },
  methods: {
    click() {
      console.log('[click]');
      imagePreviewHandler({
        selector: '#imagePreviewRef',
        images: [this.src],
      }).then(() => {
      });
    },
    getIndex(a, b) {
      return `${a}-${b}`;
    },
  },
};
</script>
<style scoped lang="scss">
.demo-grid {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.image-title {
  text-align: center;
  width: 100px;
  margin-top: 16px;
  color: #646566;
  margin-bottom: 10px;

  &--long {
    width: 132px;
  }

  &--gap {
    margin-bottom: 32px;
  }
}
</style>
