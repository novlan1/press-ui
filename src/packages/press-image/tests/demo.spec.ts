import { snapshotDemo } from '../../../../tests/unit/demo';
import Demo from '../demo.vue';

jest.mock('swiper-next/es/src/components/swiper', () => ({
  Swiper: {},
}));
jest.mock('swiper-next/es/src/components/swiper-item', () => ({
  SwiperItem: {},
}));
jest.mock('swiper-next/dist/index.css');

snapshotDemo(Demo);
