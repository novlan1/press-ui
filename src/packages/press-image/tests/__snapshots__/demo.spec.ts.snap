// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders demo correctly 1`] = `
<div class="demo-wrap">
  <div>
    <!---->
  </div>
  <div>
    <div class="press-row">
      <div class="press-col press-col--8">
        <div class="demo-grid">
          <!----> <span class="image-title image-title--gap">aspectFit</span>
        </div>
      </div>
      <div class="press-col press-col--8">
        <div class="demo-grid">
          <!----> <span class="image-title image-title--gap">aspectFill</span>
        </div>
      </div>
      <div class="press-col press-col--8">
        <div class="demo-grid">
          <!----> <span class="image-title">center</span>
        </div>
      </div>
    </div>
    <div class="press-row">
      <div class="press-col press-col--8">
        <div class="demo-grid">
          <!----> <span class="image-title image-title--gap">widthFix</span>
        </div>
      </div>
      <div class="press-col press-col--16">
        <div class="demo-grid">
          <!----> <span class="image-title image-title--long">heightFix</span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="press-row">
      <div class="press-col press-col--8">
        <div class="demo-grid">
          <!----> <span class="image-title image-title--gap">aspectFit</span>
        </div>
      </div>
      <div class="press-col press-col--8">
        <div class="demo-grid">
          <!----> <span class="image-title image-title--gap">aspectFill</span>
        </div>
      </div>
      <div class="press-col press-col--8">
        <div class="demo-grid">
          <!----> <span class="image-title">center</span>
        </div>
      </div>
    </div>
    <div class="press-row">
      <div class="press-col press-col--8">
        <div class="demo-grid">
          <!----> <span class="image-title image-title--gap">widthFix</span>
        </div>
      </div>
      <div class="press-col press-col--16">
        <div class="demo-grid">
          <!----> <span class="image-title image-title--long">heightFix</span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="press-row">
      <div class="press-col press-col--8">
        <div class="demo-grid">
          <!----> <span class="image-title">默认提示</span>
        </div>
      </div>
      <div class="press-col press-col--8">
        <div class="demo-grid">
          <!----> <span class="image-title">自定义提示</span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="press-row">
      <div class="press-col press-col--8">
        <div class="demo-grid">
          <!----> <span class="image-title">默认提示</span>
        </div>
      </div>
      <div class="press-col press-col--8">
        <div class="demo-grid">
          <!----> <span class="image-title">自定义提示</span>
        </div>
      </div>
    </div>
  </div>
  <div class="press-overlay-index">
    <!---->
  </div>
</div>
`;
